src/libraries/Common/src/System/Net/SocketAddress.cs

```diff
@@ -93,11 +93,32 @@ public SocketAddress(AddressFamily family, int size)
             }
 
             InternalSize = size;
-            Buffer = new byte[(size / IntPtr.Size + 2) * IntPtr.Size];
+
+            // TODO: workaround for https://github.com/dotnet/runtime/issues/33583
+            //       remove when Mono has pinned arrays
+            try
```

Comment:
> These SocketAddresses are constructed all over the place.

If that is the case, it would be good to use pinned arrays here.
And, yes, it is better if Mono issue is addressed in a better way then.

Reply:
Turns out adding support for pinning allocations in Mono is relatively easy.
The hack will not be needed once https://github.com/dotnet/runtime/pull/34398 is merged.

Reply:
Mono has merged the support for POH allocations. The workaround is no longer needed.